tests/currency_network/test_gas_costs_currency_network.py

```diff
@@ -91,40 +149,68 @@ def test_cost_transfer_3_mediators(
     contract = currency_network_contract_with_trustlines
     A, B, C, D, E, *rest = accounts
     tx_hash = contract.functions.transfer(50, 8, [A, B, C, D, E], EXTRA_DATA).transact(
-        {"from": A}
+        {"from": A, "gas": GAS_LIMIT_TRANSFER_3_MEDIATORS}
     )
+    assert_transaction_successful(web3, tx_hash)
     gas_cost = get_gas_costs(web3, tx_hash)
-    report_gas_costs(table, "3 hop transfer", gas_cost, limit=101_500)
+    report_gas_costs(
+        table, "3 hop transfer", gas_cost, limit=GAS_COST_TRANSFER_3_MEDIATORS
+    )
 
 
 def test_cost_first_trustline_request(web3, currency_network_contract, accounts, table):
     contract = currency_network_contract
     A, B, *rest = accounts
-    tx_hash = contract.functions.updateCreditlimits(B, 150, 150).transact({"from": A})
+    tx_hash = contract.functions.updateTrustline(
+        B, 150, 150, 1000, 1000, False
+    ).transact({"from": A, "gas": GAS_LIMIT_FIRST_TL_REQUEST})
+    assert_transaction_successful(web3, tx_hash)
     gas_cost = get_gas_costs(web3, tx_hash)
-    report_gas_costs(table, "First Trustline Update Request", gas_cost, limit=78500)
+    report_gas_costs(
+        table,
+        "First Trustline Update Request",
+        gas_cost,
+        limit=GAS_COST_FIRST_TL_REQUEST,
+    )
 
 
 def test_cost_second_trustline_request(
     web3, currency_network_contract, accounts, table
 ):
     contract = currency_network_contract
     A, B, *rest = accounts
-    contract.functions.updateCreditlimits(B, 149, 149).transact({"from": A})
-    tx_hash = contract.functions.updateCreditlimits(B, 150, 150).transact({"from": A})
+    contract.functions.updateTrustline(B, 149, 149, 1000, 1000, False).transact(
+        {"from": A}
+    )
+    tx_hash = contract.functions.updateTrustline(
+        B, 150, 150, 2000, 2000, False
+    ).transact({"from": A, "gas": GAS_LIMIT_SECOND_TL_REQUEST})
+    assert_transaction_successful(web3, tx_hash)
     gas_cost = get_gas_costs(web3, tx_hash)
-    report_gas_costs(table, "Second Trustline Update Request", gas_cost, limit=47000)
+    report_gas_costs(
+        table,
+        "Second Trustline Update Request",
+        gas_cost,
+        limit=GAS_COST_SECOND_TL_REQUEST,
+    )
 
 
 def test_cost_first_trustline(web3, currency_network_contract, accounts, table):
     contract = currency_network_contract
     A, B, *rest = accounts
     assert contract.functions.creditline(A, B).call() == 0
-    contract.functions.updateCreditlimits(B, 150, 150).transact({"from": A})
-    tx_hash = contract.functions.updateCreditlimits(A, 150, 150).transact({"from": B})
+
+    contract.functions.updateTrustline(B, 150, 150, 1000, 1000, False).transact(
```

Comment:
is this still the very first trustline? Because before we were using setAccount, but now you use already updateTrustline in the setup phase.

Reply:
No, we use `updateTrustline` in the setup phase of `currency_network_contract_with_trustlines` but this test uses `currency_network_contract`